packages/btc/src/query/offline-source.ts

```diff
@@ -0,0 +1,152 @@
+import { BtcApiTransaction, BtcApiUtxoParams, BtcAssetsApi } from '@rgbpp-sdk/service';
+import { BaseOutput, Output, Utxo } from '../transaction/utxo';
+import { NetworkType } from '../preset/types';
+import { ErrorCodes, TxBuildError } from '../error';
+import { TxAddressOutput } from '../transaction/build';
+import { isOpReturnScriptPubkey } from '../transaction/embed';
+import { getAddressType } from '../address';
+import { remove0x } from '../utils';
+import { DataSource } from './source';
+
+// * user-specific data
+export interface OfflineBtcData {
+  // Transactions whose ouputs safeguard the ownership of the RGB++ assets.
+  // These transactions are used to construct the RGB++ UTXOs to be consumed.
+  txs: BtcApiTransaction[];
+  // Extra UTXOs for paying transaction fees.
+  // Ensure these UTXOs are not used for locking RGB++ assets,
+  // as doing so will result in the loss of the RGB++ assets.
+  feeUtxos: Utxo[];
+}
+
+// * When using offline mode, the BTC fee rate must be provided,
+// * as it is not possible to query the fee rate from third-party services.
+// * However, `feeRate` is still optional at the API level to maintain compatibility.
+export class OfflineDataSource extends DataSource {
```

Comment:
I feel like it would be better to implement an `OfflineBtcAssetsApi` instead of `OfflineDataSource`. 

Here are my reasons, please let me know how you guys think:

1. Essentially, `DataSource` queries data from `BtcAssetsApi` and performs some filtering and transformation before returning it to the caller. Since it contains logic, it would require more debugging and testing on `OfflineDataSource`.
2. On the API stability level, `BtcAssetsApi` is considered more stable than `DataSource`, as it only requests data from `btc-assets-api` without additional actions, while `DataSource` contains logic and can undergo small optimizations from time to time. This means that when we update `DataSource`, we also have to check `OfflineDataSource` to see if anything needs extra testing.

Reply:
Or, we don't extend from `BtcAssetsApi` at all and allow passing an offline source to `DataSource`:

```ts
constructor(service: BtcAssetsApi | OfflineBtcDataSource, networkType: NetworkType) { ... }
```

Reply:
I support that. Implement the same interface as BtcAssetsApi and pass it, which is less destructive.

Reply:
I second that. Implement the same interface as BtcAssetsApi and pass it, which is better and has fewer modifications.

Reply:
Implemented `OfflineBtcAssetsDataSource` that extends `BtcAssetsApi` in e9e950b9df57fd35e3ac85c0689a83dac51a7b67. Please let me know your thoughts.